### PR TITLE
Update Dokka configuration

### PR DESCRIPTION
**Pull request by zsmb13**
https://stream-io.atlassian.net/browse/CAS-654 (pt1, fixing Dokka output)

### Description

Updates Dokka to a development version where inherited members can be listed on separate tabs.

To test, run `gradlew dokkaHtmlMultimodule`. Output is in `/build/dokka/htmlMultiModule` - just open `index.html` there.

image

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] ~Changelog updated with client-facing changes~
- [ ] ~New code is covered by unit tests~
- [x] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

File: scripts/publish-mavencentral.gradle [hunk 1]
```diff
@@ -13,6 +13,12 @@ task androidSourcesJar(type: Jar) {
     }
 }
 
+tasks.withType(dokkaHtmlPartial.getClass()).configureEach {
+    pluginsMapConfiguration.set(
+            ["org.jetbrains.dokka.base.DokkaBase": """{ "separateInheritedMembers": true}"""]
+    )
```
Comment (zsmb13) [inline scripts/publish-mavencentral.gradle:19]: Undocumented black magic, see https://kotlinlang.slack.com/archives/C0F4UNJET/p1612961697050300

File: gradle.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 android.enableJetifier=true
 android.useAndroidX=true
-org.gradle.jvmargs=-Xmx1536m
+org.gradle.jvmargs=-Xmx2048m
```
Comment (zsmb13) [inline gradle.properties:3]: Dokka likes lots of RAM, raised this a bit to prevent process crashes

File: stream-chat-android-core/build.gradle [hunk 3]
```diff
@@ -4,7 +4,6 @@ plugins {
     id 'java-library'
     id 'kotlin'
     id "com.github.dcendents.android-maven"
-    id "org.jetbrains.dokka"
```
Comment (zsmb13) [inline stream-chat-android-core/build.gradle:7]: Not needed anymore, applied in the publishing script for each module

File: build.gradle [hunk 4]
```diff
@@ -5,17 +5,17 @@ apply plugin: "com.github.ben-manes.versions"
...
 
 buildscript {
     repositories {
         maven { url "https://plugins.gradle.org/m2/" }
         google()
         mavenCentral()
-
-        jcenter() {
+        maven {
+            url 'https://maven.pkg.jetbrains.space/kotlin/p/dokka/dev'
```
Comment (JcMinarro) [inline build.gradle:16]: :partying_face:

File: buildSrc/src/main/kotlin/com/getstream/sdk/chat/Dependencies.kt [hunk 5]
```diff
@@ -28,7 +28,7 @@ private const val CONSTRAINT_LAYOUT_VERSION = "2.0.3"
 private const val COROUTINES_VERSION = "1.4.1"
 private const val DEXTER_VERSION = "6.2.1"
 private const val DRAWABLETOOLBOX_VERSION = "1.0.7"
-private const val DOKKA_VERSION = "1.4.20"
+private const val DOKKA_VERSION = "1.4.20.2-dev-62"
```
Comment (JcMinarro) [inline buildSrc/src/main/kotlin/com/getstream/sdk/chat/Dependencies.kt:31]: Isn't it working on a stable version yet?

**Review by JcMinarro — APPROVED** [review]
LGTM

File: buildSrc/src/main/kotlin/com/getstream/sdk/chat/Dependencies.kt (hunk 5 above)
Comment (zsmb13) [reply buildSrc/src/main/kotlin/com/getstream/sdk/chat/Dependencies.kt:31]: Not on stable, not even on RC 😅. Only in the development versions for now.

**Pull request merged**